### ISSUE TITLE
Add a vignette to the R package

### ISSUE THREAD

**Issue by kevinrue** [beringresearch/ivis#38]
Hello,

For [JOSS review](https://github.com/openjournals/joss-reviews/issues/1596).

**Is your feature request related to a problem? Please describe.**

The R package lacks documentation of an application to a real-life dataset.

**Describe the solution you'd like**

Please add a [vignette](http://r-pkgs.had.co.nz/vignettes.html) in the R package demonstrating at least an example application to a single-cell dataset. Basically, the equivalent of the scanpy workflow [here](https://bering-ivis.readthedocs.io/en/latest/scanpy_singlecell.html).

A convenient way to use the `pbmc3k` dataset for demonstration purposes is the Bioconductor [TENxPBMCData](https://bioconductor.org/packages/release/data/experiment/html/TENxPBMCData.html) package.

Suggested code:
```r
library(TENxPBMCData)
tenx_ pbmc3k <- TENxPBMCData(dataset = "pbmc3k")
```

Ideally, consider using the vignette (or a separate one) to also give an introduction to the functionality of the R package.
It is not necessary to duplicate information already described in the documentation of the Python package (DRY principle); you may simply include a link to the [main page](https://bering-ivis.readthedocs.io/en/latest/index.html).

**Describe alternatives you've considered**

A working example of an R workflow could also be included in the documentation of the Python package, although this is probably unnecessarily difficult to maintain.
Ideally, that example would be run and tested for every new release of the Python and R source code.

**Additional context**
Once you have an R vignette written, you should also consider using [pkgdown](https://pkgdown.r-lib.org) to automatically create a GitHub website including the full package documentation.

**Comment by idroz** [comment]
Hi there - I've added an [R vignette](https://github.com/beringresearch/ivis/tree/master/R-package/vignettes) that demonstrates how `ivis` can be integrated into a Seurat workflow. The vignette uses an example single cell dataset mentioned earlier.

With respect to pkgdown, as the main package implementation is in python, we feel that readthedocs is more appropriate for the project. As documentation expands, we will incorporate R vignette into the readthedocs builds.

**Comment by kevinrue** [comment]
Thanks!

I could run the R vignette manually, but that required following it on your [repository](https://github.com/beringresearch/ivis/blob/master/R-package/vignettes/ivis_singlecell.Rmd#L19), and manually running the chunk of bash code to download the raw data from 10X (https://github.com/beringresearch/ivis/blob/master/R-package/vignettes/ivis_singlecell.Rmd#L19).

One of the main advantages of an R vignette is that it is generally compiled and installed with the R package itself (i.e., `browseVignettes("ivis")` should display the vignettes available).
This allows users to preview the package functionality before installing it, but also demonstrates that the package simply works.

To do so, as long as the R package is distributed directly from GitHub, I suggest that you update the installation command to the following, in order to build the vignette on the users' machine.
```r
devtools::install_github("beringresearch/ivis/R-package", build_vignettes = TRUE)
```

That said, the command above currently crashes because the vignette includes a chunk of code that needs to be run manually to  download the demonstration dataset (https://github.com/beringresearch/ivis/blob/master/R-package/vignettes/ivis_singlecell.Rmd#L19)

I will shortly open a PR to demonstrate the use of `TENxPBMCData` (mentioned above).

Having tested it locally, I must point out that installing the R package with `build_vignettes = TRUE` mentioned above will take longer to run, as it basically needs to run the method on the PBMC 3k dataset.
I haven't timed it precisely, but the overall process took a few minutes on my laptop.



I suggested `pkgdown` only because it automatically compiles and serves the R man pages and vignettes for free every time the R package is updated.
That said, you probably want to recompile the doc whenever any one of the Python or R package is updated, so I agree that readthedocs is more appropriate as the central documentation for this project.

If you can also compile and serve the R vignette there, I've got nothing to add.
That said, readthedocs is probably the best solution to update the documentation

**Comment by idroz** [comment]
Hi! Really appreciate the feedback on this and the pull request. Will do the checks and update!

**Comment by idroz** [comment]
Works great - merged #42 into master - thank you!

**Comment by kevinrue** [comment]
With #43 I think that we can call this issue resolved.
